File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (this.isNothing(input))  { return "Fine. Be that way!"; }
  if (this.isYelling(input))  { return "Whoa, chill out!"; }
  if (this.isQuestion(input)) { return "Sure."; }
  return "Whatever.";
};

Bob.prototype.isQuestion = function(input) {
  return input.endsWith("?");
};

Bob.prototype.isNothing = function(input) {
  return input.trim().length == 0;
};

Bob.prototype.isYelling = function(input) {
  return input.toUpperCase() == input && input.match(/[A-z]/u);
};

module.exports = Bob;

File: python/bob/bob.py
import re

def hey(phrase):
    phrase = phrase.strip()

    if is_yelling(phrase):  return 'Whoa, chill out!'
    if is_question(phrase): return 'Sure.'
    if is_nothing(phrase):  return 'Fine. Be that way!'
    return 'Whatever.'

def is_yelling(phrase):
    return phrase.upper() == phrase and re.search('[A-Z]', phrase)

def is_question(phrase):
    return phrase.endswith('?')

def is_nothing(phrase):
    return len(phrase) == 0
